pkg/ddl/column_type_change_test.go

```diff
@@ -498,31 +498,52 @@ func TestChangingColOriginDefaultValueAfterAddColAndCastFail(t *testing.T) {
 		if job.SchemaState == model.StateWriteOnly || job.SchemaState == model.StateWriteReorganization {
 			tbl := external.GetTableByName(t, tk, "test", "t")
 			if len(tbl.WritableCols()) != 4 {
-				errMsg := fmt.Sprintf("cols len:%v", len(tbl.WritableCols()))
+				errMsg := fmt.Sprintf("job ID:%d, cols len:%v", job.ID, len(tbl.WritableCols()))
 				checkErr = errors.New("assert the writable column number error" + errMsg)
 				return
 			}
-			originalDV := fmt.Sprintf("%v", tbl.WritableCols()[3].OriginDefaultValue)
-			expectVal := "0000-00-00 00:00:00"
-			if originalDV != expectVal {
-				errMsg := fmt.Sprintf("expect: %v, got: %v", expectVal, originalDV)
-				checkErr = errors.New("assert the write only column origin default value error" + errMsg)
-				return
+			// modify column x
+			if job.ID == 107 {
+				originalDV := fmt.Sprintf("%v", tbl.WritableCols()[3].OriginDefaultValue)
+				expectVal := "0000-00-00 00:00:00"
+				if originalDV != expectVal {
+					errMsg := fmt.Sprintf("job ID:%d, expect: %v, got: %v", job.ID, expectVal, originalDV)
+					checkErr = errors.New("assert the write only column origin default value error" + errMsg)
+					return
+				}
+				// The cast value will be inserted into changing column too.
+				_, err := tk1.Exec("UPDATE t SET a = '18apf' WHERE x = '' AND a = 'mul'")
+				if err != nil {
+					checkErr = err
+					return
+				}
 			}
-			// The casted value will be inserted into changing column too.
-			_, err := tk1.Exec("UPDATE t SET a = '18apf' WHERE x = '' AND a = 'mul'")
-			if err != nil {
-				checkErr = err
-				return
+			// modify column b
+			if job.ID == 108 {
```

Comment:
If we add some ddl jobs in bootstrap, the job ID will change. Maybe get the prev job ID and use expression like `job.ID == prevJobID + 1` is better.